packages/react/src/host/controller.tsx

```diff
@@ -1,16 +1,45 @@
-import type {ComponentType} from 'react';
-import type {RemoteComponentType} from '@remote-ui/core';
+import type {ComponentType, ReactElement} from 'react';
+import type {
+  Controller,
+  RemoteComponentProps,
+  RemoteTextProps,
+  RenderComponentOptions,
+  Renderer,
+  RenderTextOptions,
+} from './types';
+
+import {renderComponent as defaultRenderComponent} from './RemoteComponent';
+import {renderText as defaultRenderText} from './RemoteText';
 
 export interface ComponentMapping {
   [key: string]: ComponentType<any>;
 }
 
-export interface Controller {
-  get(type: string | RemoteComponentType<string, any, any>): ComponentType<any>;
+interface RendererFactory {
+  renderComponent(
+    props: RemoteComponentProps,
```

Comment:
I still really wish the component itself was alone as the first argument of these functions, since you will always need to pull that off this argument, but can't do it by destructuring if you end up needing to call `renderDefault()`...

Reply:
`renderDefault` could in theory be made to take no arguments, couldn't it?

Reply:
Yes, we could. 👍 James is making a change